src/octopath.cpp

```diff
@@ -106,199 +153,338 @@ bool OctoClass::OctoRRG(const Eigen::Vector3d &p0,
+      *n_rrg_nodes = n_nodes;
+      *plan_time = (ros::Time::now() - t0).toSec();
+      return false;
+    } else {
+      for (uint i = 0; i < index_path.size(); i++) {
+        sol_path.push_back(obj_rrg.rrgraph_.nodes_[index_path[i]].pos_);
+      }
+      // ROS_INFO("Path size: %d", static_cast<int>(sol_path.size()));
+    }
+  } else {
+    ROS_WARN("[mapper] RRG Error: Could not find a path from p0 to pf!");
+    *n_rrg_nodes = n_nodes;
+    *plan_time = (ros::Time::now() - t0).toSec();
+    return false;
+  }
 
-        // Get RRG parameter
-        n_nodes = obj_rrg.rrgraph_.n_nodes_;
-        const double max_dist = std::min(gamma*pow(log(n_nodes)/n_nodes,
-                                        dim_inv), steer_param);
-
-        // Get nodes within max_dist radius
-        std::vector<uint> near_nodes;
-        std::vector<double> costs;
-        obj_rrg.NodesWithinRadius(max_dist, sample, &near_nodes, &costs);
-
-        // Add non-colliding nodes within max_dist radius
-        for (uint i = 0; i < near_nodes.size(); i++) {
-            neighbor_pos = obj_rrg.rrgraph_.nodes_[near_nodes[i]].pos_;
-            if (free_space_only) {
-                col_check = CheckCollision(neighbor_pos, sample);
-            } else {
-                col_check = CheckOccupancy(neighbor_pos, sample);
-            }
-
-            if (col_check != 1) {
-                obj_rrg.AddEdge(index, near_nodes[i], costs[i]);
-            }
-        }
+  // Prune results if requested
+  if (prune_result) {
+    this->PathPruning(sol_path, free_space_only, path);
+  } else {
+    *path = sol_path;
+  }
 
-        // Go back to 'while' if there is a connection between p0 and pf
-        if (connected_graph) {
-            continue;
-        }
+  // Populate time and nodes
+  *n_rrg_nodes = n_nodes;
+  *plan_time = (ros::Time::now() - t0).toSec();
 
-        // Check if new node connects with the final destination
-        // This portion does not need to execute after one path has been found between p0 and pf
-        cost = obj_rrg.DistanceToNode(index, pf);
-        if (cost <= max_dist) {
-            if (free_space_only) {
-                col_check = CheckCollision(pf, sample);
-            } else {
-                col_check = CheckOccupancy(pf, sample);
-            }
-
-            if (col_check != 1) {
-                obj_rrg.AddNode(pf, &final_index);
-                obj_rrg.AddEdge(index, final_index, cost);
-                connected_graph = true;
-                ROS_INFO("[mapper] Found connection to destination!");
-            }
-        }
-    }
+  // ROS_INFO("[mapper] nNodes: %d", static_cast<int>(obj_rrg.rrgraph_.n_nodes_));
+  // ROS_INFO("[mapper] nEdges: %d", static_cast<int>(obj_rrg.rrgraph_.n_edges_));
+
+  return true;
+}
 
-    // Publish graph into Rviz if requested
-    if (publish_rviz) {
-        obj_rrg.rrgraph_.GraphVisualization(graph_markers);
+bool OctoClass::Astar(const octomap::point3d &p0,
```

Comment:
This is the cherry on top of this PR...
This is not necessarily easy to read or understand. If you really want to understand this, I suggest looking into [A* implementations](https://en.wikipedia.org/wiki/A*_search_algorithm)